core/build.gradle.kts

```diff
@@ -53,8 +53,10 @@ kotlin {
         val jvmMain by getting
         val jvmTest by getting
 
-        createSourceSet("nativeMain", parent = commonMain, children = nativeTargets)
-        createSourceSet("nativeTest", parent = commonTest, children = nativeTargets)
+        val nativeMain = createSourceSet("nativeMain", parent = commonMain, children = nativeTargets)
+        val nativeTest = createSourceSet("nativeTest", parent = commonTest, children = nativeTargets)
+        createSourceSet("appleMain", parent = nativeMain, children = appleTargets)
+        createSourceSet("appleTest", parent = nativeTest, children = appleTargets)
```

Comment:
BTW, the same change is needed in the bytestring's `build.gradle.kts`. There are no native sources to cause issues right now, but it would be nice to prevent them from appearing in the future.